packages/core/contracts/oracle/implementation/SkinnyOptimisticOracle.sol

```diff
@@ -0,0 +1,716 @@
+        return _settle(_requester, _identifier, _timestamp, _ancillaryData, _request);
+    }
+
+    /**
+     * @notice Computes the current state of a price request. See the State enum for more details.
+     * @param _requester sender of the initial price request.
+     * @param _identifier price identifier to identify the existing request.
+     * @param _timestamp timestamp to identify the existing request.
+     * @param _ancillaryData ancillary data of the price being requested.
+     * @param _request price request parameters.
+     * @return the State.
+     */
+    function getState(
+        address _requester,
+        bytes32 _identifier,
+        uint256 _timestamp,
+        bytes memory _ancillaryData,
+        Request memory _request
+    ) external override nonReentrant() returns (OptimisticOracleInterface.State) {
+        return _getState(_requester, _identifier, _timestamp, _ancillaryData, _request);
+    }
+
+    /**
+     * @notice Checks if a given request has resolved, expired or been settled (i.e the optimistic oracle has a price).
+     * @param _requester sender of the initial price request.
+     * @param _identifier price identifier to identify the existing request.
+     * @param _timestamp timestamp to identify the existing request.
+     * @param _ancillaryData ancillary data of the price being requested.
+     * @param _request price request parameters. The hash of these parameters must match with the request hash that is
+     * associated with the price request unique ID {requester, identifier, timestamp, ancillaryData}, or this method
+     * will revert.
+     * @return boolean indicating true if price exists and false if not.
+     */
+    function hasPrice(
+        address _requester,
+        bytes32 _identifier,
+        uint256 _timestamp,
+        bytes memory _ancillaryData,
+        Request memory _request
+    ) public override nonReentrant() returns (bool) {
+        bytes32 requestId = _getId(_requester, _identifier, _timestamp, _ancillaryData);
+        _validateRequestHash(requestId, _request);
+        OptimisticOracleInterface.State state =
+            _getState(_requester, _identifier, _timestamp, _ancillaryData, _request);
+        return
+            state == OptimisticOracleInterface.State.Settled ||
+            state == OptimisticOracleInterface.State.Resolved ||
+            state == OptimisticOracleInterface.State.Expired;
+    }
+
+    /**
+     * @notice Generates stamped ancillary data in the format that it would be used in the case of a price dispute.
+     * @param _ancillaryData ancillary data of the price being requested.
+     * @param _requester sender of the initial price request.
+     * @return the stamped ancillary bytes.
+     */
+    function stampAncillaryData(bytes memory _ancillaryData, address _requester)
+        public
+        pure
+        override
+        returns (bytes memory)
+    {
+        return _stampAncillaryData(_ancillaryData, _requester);
+    }
+
```

Comment:
do you want to add a comment block to show you are moving into private functions?
```suggestion
    /****************************************
     *    PRIVATE AND INTERNAL FUNCTIONS    *
     ****************************************/
```